lib/docker.ts

```diff
@@ -153,40 +234,45 @@ export const getContainerIP = async function (
 // Until we can properly configure the baseUrl retry intervals,
 // We need to make sure the server is already running before cypress
 // https://github.com/cypress-io/cypress/issues/22676
-export const waitOnNextcloud = async function (ip: string) {
+export const waitOnNextcloud = async function(ip: string) {
 	console.log('├─ Waiting for Nextcloud to be ready... ⏳')
 	await waitOn({ resources: [`http://${ip}/index.php`] })
 	console.log('└─ Done')
 }
 
-const runExec = async function (
+const runExec = async function(
 	container: Docker.Container,
 	command: string[],
-	verbose: boolean = false
+	verbose = false,
+	user = 'www-data'
 ) {
 	const exec = await container.exec({
 		Cmd: command,
 		AttachStdout: true,
 		AttachStderr: true,
-		User: 'www-data',
+		User: user,
 	})
 
-	return new Promise((resolve, reject) => {
+	return new Promise<string>((resolve, reject) => {
 		exec.start({}, (err, stream) => {
 			if (stream) {
-				stream.setEncoding('utf-8')
-				stream.on('data', str => {
-					if (verbose && str.trim() !== '') {
-						console.log(`├─ ${str.trim().replace(/\n/gi, '\n├─ ')}`)
+				const data = [] as string[]
+				stream.setEncoding('utf8')
+				stream.on('data', (str) => {
+					data.push(str)
+					const printable = str.replace(/\p{C}/gu, '').trim()
```

Comment:
Please explain this :)

Reply:
Outdated as I had to fix this function. I had to handle the stdout / stderr stream differently.